src/odemis/driver/xt_client.py

```diff
@@ -2158,10 +2163,10 @@ def _setBeamletIndex(self, beamlet_index):
         return tuple(int(i) for i in new_beamlet_index)  # convert tuple values to integers.
 
     def _setMultiBeamMode(self, multi_beam_mode):
-        # TODO: When changing the beam mode of the microscope we don't want to also change the aperture and beamlet
-        #  index. However, it this is the current implementation direction of TFS. Therefore in the future code like
-        #  the uncommented parts below need to be implemented. Note this code still needs to be tested and is
-        #  therefore now just an example of an implementation.
+        # TODO: When changing the beam mode of the microscope changes we don't want to also change the aperture and
+        #  beamlet index. However, it seems that this is the way TFS will be implementing and supporting XTtoolkit.
+        #  Therefore in the future code like the uncommented parts below need to be implemented.
+        #  Note: this code still needs to be tested and is therefore now just an example of an implementation.
```

Comment:
BTW, I've just learned that TFS has new plans for the API. At least, when going back to single beam mode, the next version  of their software will restore the previous beamlet index. Aperture might also be automatic.
Let's see...